Legacy/bonej/src/main/java/org/bonej/plugins/ConnectedComponents.java

```diff
@@ -115,13 +115,20 @@ public ConnectedComponents() {
 		final int nProcessors = Runtime.getRuntime().availableProcessors();
 		final int minSlicesPerChunk = 10;
 
-		// set up number of chunks
-		final int nChunks = nSlices < minSlicesPerChunk * nProcessors
-				? (int) Math.ceil((double) nSlices / (double) minSlicesPerChunk)
-				: nProcessors;
-		// set up chunk sizes - last chunk is the remainder
-		final int slicesPerChunk = (int) Math.ceil((double) nSlices / (double) nChunks);
-
+		// set up number of chunks and chunk sizes
+		int nChunks = 1;
+		int slicesPerChunk = nSlices;
+		if (nSlices < minSlicesPerChunk) {
+			slicesPerChunk = nSlices;
+			nChunks = 1;
+		} else if (nSlices <= minSlicesPerChunk * nProcessors) {
+			slicesPerChunk = minSlicesPerChunk;
+			nChunks = (int) Math.ceil((double) nSlices / (double) minSlicesPerChunk); 
+		} else if (nSlices > minSlicesPerChunk * nProcessors) {
+			nChunks = nProcessors;
+			slicesPerChunk = (int) Math.floor((double) nSlices / (double) nChunks);
+		}
```

Comment:
This comment can be ignored, but I wanted to flag that it was unintuitive for me that the first two `if` clauses assign `slicePerChunk` first, and the third clause does the opposite, just in terms of the flow of reading a piece of code I had never looked at before. Might be worth swapping the order in the first two clauses. But again, happy to leave it :)